test-framework/junit5/src/main/java/io/quarkus/test/junit/internal/CustomMapEntryConverter.java

```diff
@@ -0,0 +1,55 @@
+package io.quarkus.test.junit.internal;
+
+import java.util.AbstractMap;
+import java.util.Map;
+import java.util.Set;
+
+import com.thoughtworks.xstream.converters.MarshallingContext;
+import com.thoughtworks.xstream.converters.UnmarshallingContext;
+import com.thoughtworks.xstream.converters.collections.MapConverter;
+import com.thoughtworks.xstream.io.HierarchicalStreamReader;
+import com.thoughtworks.xstream.io.HierarchicalStreamWriter;
+import com.thoughtworks.xstream.mapper.Mapper;
+
+/**
+ * A custom Map.Entry converter that always uses AbstractMap.SimpleEntry for unmarshalling.
+ * This is probably not semantically correct 100% of the time, but it's likely fine
+ * for all the cases where we are using marshalling / unmarshalling.
+ *
+ * The reason for doing this is to avoid XStream causing illegal access issues
+ * for internal JDK types
+ */
+@SuppressWarnings({ "rawtypes", "unchecked" })
+public class CustomMapEntryConverter extends MapConverter {
+
+    private final Set<String> SUPPORTED_CLASS_NAMES = Set
+            .of(Map.entry(Integer.MAX_VALUE, Integer.MAX_VALUE).getClass().getName());
+
+    public CustomMapEntryConverter(Mapper mapper) {
+        super(mapper);
+    }
+
+    @Override
+    public boolean canConvert(Class type) {
+        return (type != null) && SUPPORTED_CLASS_NAMES.contains(type.getName());
```

Comment:
Wondering if `Map.Entry.class.isAssignableFrom(type)` could work here

Reply:
It would, but we want to limit this as much as possible to cases that we know don't work otherwise